src/main/resources/assets/opencomputers/loot/openos/lib/serialization.lua

```diff
@@ -1,134 +1,160 @@
-  end
-  local ok, output = pcall(result)
-  if not ok then
-    return nil, output
-  end
-  return output
+function serialization.serializeToFile(path, array, prettyLook, indentationWidth, indentUsingTabs, appendToFile)
+	checkArg(1, path, "string")
+	checkArg(2, array, "table")
+	filesystem.makeDirectory(filesystem.path(path) or "")
+	local file = io.open(path, appendToFile and "a" or "w")
+	file:write(serialization.serialize(array, prettyLook, indentationWidth, indentUsingTabs))
+	file:close()
 end
 
+function serialization.unserializeFromFile(path)
+	checkArg(1, path, "string")
+	if filesystem.exists(path) then
+		if filesystem.isDirectory(path) then
+			error("\"" .. path .. "\" is a directory")
+		else
+			local file = io.open(path, "r")
+			local data = serialization.unserialize(file:read("*a"))
+			file:close()
+			return data
+		end
+	else
+		error("\"" .. path .. "\" doesn't exists")
+	end
+end
+
+------------------------------------------------- Playground -----------------------------------------------------------------
+
+-- local sampleTable = {
+-- 	"one",
+-- 	"two",
+-- 	"three",
+-- 	[9] = 9,
+-- 	[10] = "ten",
+-- 	[11] = "eleven",
+-- 	["test"] = "sucess",
+-- 	["tableTest"] = {
+-- 		["tableInTable"] = {
+-- 			[12] = "blah-blah",
+-- 			["hello"] = "world",
+-- 		},
+-- 		"cyka",
+-- 		"blyad",
+-- 	},
+-- 	["nilTest"] = nil,
+-- 	["booleanTest"] = true,
+-- }
+
+-- local function fillTable(count)
+-- 	print("Filling table with " .. count .. " random values...")
+-- 	for i = 1, count do
+-- 		table.insert(sampleTable, math.random(1, 100000))
+-- 	end
+-- end
+
+-- local function executeAndPrintExecutionTime(method, ...)
+-- 	local oldTime = os.clock()
+-- 	method(...)
+-- 	print("Execution time: " .. os.clock() - oldTime .. " seconds")
+-- end
+
+-- local function compareAlgorithms()
+-- 	print(" ")
+-- 	fillTable(10000)
+-- 	print(" ")
+-- 	local oldSerialization = require("openOSSerialization")
+-- 	print("Old:")
+-- 	executeAndPrintExecutionTime(oldSerialization.serialize, sampleTable)
+-- 	print("New:")
+-- 	executeAndPrintExecutionTime(serialization.serialize, sampleTable)
+-- 	print(" ")
+-- end
+
+-- compareAlgorithms()
+
+----------------------------------------------------------------------------------------------------------------------
+
+-- print("Serializing...")
+-- local serializedTable = serialization.serialize(sampleTable, true)
+-- print(" ")
+-- print(serializedTable)
+-- print(" ")
+-- print("Done.")
+-- print("Unserializing...")
+-- local unserializedTable = serialization.unserialize(serializedTable)
+-- print("Done.")
+-- print("Saving to file...")
+-- serialization.serializeToFile("serializationTest.lua", sampleTable, true)
+-- print("Done.")
+-- print("Loading from file...")
+-- serialization.unserializeFromFile("serializationTest.lua")
+-- print("Done.")
+
+----------------------------------------------------------------------------------------------------------------------
+
```

Comment:
Remove this comment playground section, please. Feel free to share tests via gists and give links to those gists in the pr discussion.